File: scripts/clean-coreConditions.js
const fs = require('fs');
const path = require('path');

const p = path.join(__dirname, '..', 'static', 'js', 'data', 'coreConditions.js');
if (!fs.existsSync(p)) {
  console.error('File not found:', p);
  process.exit(2);
}

let s = fs.readFileSync(p, 'utf8');

// Remove UTF-8 BOM if present
s = s.replace(/^\uFEFF/, '');

// Normalize common “smart” quotes to ASCII
s = s.replace(/[\u2018\u2019\u201A\u201B]/g, "'")
     .replace(/[\u201C\u201D\u201E\u201F]/g, '"');

// Remove other non-printable / non-UTF8-friendly control characters
// Keep common whitespace and printable Unicode (preserve accents etc.) by removing C0 controls except tab/newline/carriage-return
s = s.replace(/[\x00-\x08\x0B\x0C\x0E-\x1F]/g, '');

fs.writeFileSync(p, s, 'utf8');
console.log('Cleaned:', p);

File: scripts/clean-coreConditions.py
import io
import os
import re

ROOT = os.path.join(os.path.dirname(__file__), '..')
P = os.path.normpath(os.path.join(ROOT, 'static', 'js', 'data', 'coreConditions.js'))

if not os.path.exists(P):
    print('File not found:', P)
    raise SystemExit(2)

with io.open(P, 'r', encoding='utf-8') as f:
    s = f.read()

# Remove BOM if present
if s.startswith('\ufeff'):
    s = s.lstrip('\ufeff')

# Normalize smart quotes
s = s.replace('\u2018', "'").replace('\u2019', "'")
s = s.replace('\u201C', '"').replace('\u201D', '"')

# Remove other C0 control characters except tab/newline/carriage-return
s = re.sub(r'[\x00-\x08\x0B\x0C\x0E-\x1F]', '', s)

with io.open(P, 'w', encoding='utf-8') as f:
    f.write(s)

print('Cleaned:', P)
